sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -0,0 +1,389 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jdbc;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+import java.net.InetAddress;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.ArrayList;
+
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.testing.NeedsRunner;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.derby.drda.NetworkServerControl;
+import org.apache.derby.jdbc.ClientDataSource;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Test on the JdbcIO.
+ */
+public class JdbcIOTest implements Serializable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcIOTest.class);
+
+  private ClientDataSource dataSource;
+
+  @Before
+  public void setup() throws Exception {
+    System.setProperty("derby.stream.error.file", "target/derby.log");
+
+    NetworkServerControl derbyServer = new NetworkServerControl(InetAddress.getByName
+        ("localhost"), 1527);
+    derbyServer.start(null);
+
+    dataSource = new ClientDataSource();
+    dataSource.setCreateDatabase("create");
+    dataSource.setDatabaseName("target/beam");
+    dataSource.setServerName("localhost");
+    dataSource.setPortNumber(1527);
+
+    Connection connection = dataSource.getConnection();
+
+    Statement statement = connection.createStatement();
+    try {
+      statement.executeUpdate("create table BEAM(id INT, name VARCHAR(500))");
+      statement.executeUpdate("create table TEST(ID INT, NAME VARCHAR(200))");
+    } catch (Exception e) {
+      LOGGER.warn("Can't create table BEAM, it probably already exists", e);
+    } finally {
+      statement.close();
+    }
+
+    statement = connection.createStatement();
+    statement.executeUpdate("delete from BEAM");
+    statement.close();
+
+    String[] scientists = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday",
+        "Newton", "Bohr", "Galilei", "Maxwell"};
+    for (int i = 1; i <= 1000; i++) {
```

Comment:
This is just repeating things 1000 times; I think looping over [0, 1000) is more idiomatic than looping over [1, 1000].